spec/seafoam/passes/truffle_spec.rb

```diff
@@ -50,82 +50,84 @@
       end
 
       describe "with :simplify_truffle_args" do
-        before :each do
-          @filename = File.expand_path(
-            "../../../examples/#{graalvm_version}/ruby/example_polymorphic_receiver.bgv.gz",
-            __dir__,
-          )
-          @phase_index = "After TruffleTier"
-          @graph = Seafoam::SpecHelpers.example_graph(@filename, @phase_index)
-        end
-
-        it "substitutes TruffleArgument nodes for LoadIndexed nodes" do
-          before_load_indexed_nodes = @graph.nodes.values.select do |n|
-            n.node_class.end_with?(".compiler.nodes.java.LoadIndexedNode")
-          end.map(&:dup)
-          before_truffle_argument_nodes = @graph.nodes.values.select do |n|
-            n.props[:synthetic_class] == "TruffleArgument"
+        ["example_polymorphic_receiver", "example_arith_operator"].each do |example|
```

Comment:
So this expands to `describe { before, it, it, before, it, it }`...
which seems weird to me (I'm concerned that the contents of the second before block would overwrite the first).
Is it possible the before blocks are attached to the `it` blocks that follow (rather than just having both on the `describe`)?

Reply:
You're right. This is broken. I had confirmed that outer `graalvm_version` was correct, but `@filename` is wrong here. I'll take another look. I really wanted to avoid dynamically defining the specs, but if that's what necessary so be it.

Reply:
I wrapped everything in another `describe` block, this time printing out the context so it's easier to follow. I applied the same change to the GraalVM version stuff higher up. It makes the spec names a fair bit longer, but it actually looks quite nice in RubyMine (presumably VS Code as well).

Since everything ended up getting indented another level or two, it'd be best to enable the "ignore whitespace" option when looking at the diff.

Reply:
I always ignore whitespace changes when looking at diffs :)